src/content/docs/en/core-concepts/astro-syntax.mdx

```diff
@@ -148,6 +148,41 @@ In Astro, you use the standard `kebab-case` format for all HTML attributes inste
 <div class="box" data-value="3" />
 ```
 
+#### Node in Script Body
+
+In Astro, you can't use JSX node in script body.
+
+```astro title="TodoList.astro" "<TodoItem todo={todo} />"
+---
+import TodoItem from "./TodoItem.astro"
+const todos = [
+  { id: 1, desc: "Todo 1" },
+  { id: 2, desc: "Todo 2" }
+]
+
+// 🚫 This doesn't work
+const renderedTodos = todos.map(todo => <TodoItem todo={todo} />) 
+---
+
+<div>
+  {renderedTodos}
+</div>
+```
```

Comment:
I think we should clarify this early at the top of the `JSX-like expressions`, maybe under:

> You can define local JavaScript variables inside of the frontmatter component script between the two code fences (---) of an Astro component.

We should clarify that anything between two code fences only support JS or TS, and not JSX.